### PR TITLE
Nucleo f030r8

### LINKED ISSUE #3923: boards: provide support for Nucleo-64 F030R8
**_Reported by Maciej Dębski:_**

The aim of an issue is to provide support for development board Nucleo-64 F030R8.

(Imported from Jira ZEP-2492)

### PR DESCRIPTION
**Pull request by superna9999**
Fixes #3923
boards: provide support for Nucleo-64 F030R8

This PR is an updated version of #1103 since @MaciejDebski won't have time to maintain this PR, authorship of the patches has been preserved.

This PR contains a fix to solve the missing memory issues on the automated build test suite.

### CODE REVIEW

File: boards/arm/nucleo_f030r8/pinmux.c [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#include "pinmux/pinmux.h"
+
+#include "pinmux/stm32/pinmux_stm32.h"
+
+/* pin assignments for NUCLEO-F030R8 board */
+static const struct pin_config pinconf[] = {
+#ifdef CONFIG_UART_STM32_PORT_1
+	{STM32_PIN_PB6,  STM32F0_PINMUX_FUNC_PB6_USART1_TX},
+	{STM32_PIN_PB7, STM32F0_PINMUX_FUNC_PB7_USART1_RX},
+#endif	/* CONFIG_UART_STM32_PORT_1 */
```
Comment (erwango) [inline boards/arm/nucleo_f030r8/pinmux.c]: Not in line with doc and device tree

File: drivers/clock_control/stm32f0x_ll_clock.c [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	/*
+	 * PLL DIV
+	 * 1  -> LL_RCC_PLLSOURCE_HSE_DIV_1  -> 0x00010000
+	 * 2  -> LL_RCC_PLLSOURCE_HSE_DIV_2  -> 0x00010001
+	 * 3  -> LL_RCC_PLLSOURCE_HSE_DIV_3  -> 0x00010002
+	 * ...
+	 * 16 -> LL_RCC_PLLSOURCE_HSE_DIV_16 -> 0x0001000F
+	 */
+	pllinit->PLLDiv = (RCC_CFGR_PLLSRC_HSE_PREDIV |
+					(CONFIG_CLOCK_STM32_PLL_PREDIV1 - 1));
```
Comment (erwango) [inline drivers/clock_control/stm32f0x_ll_clock.c]: CLOCK_STM32_PLL_PREDIV1 is not available in Kconfig.
If it should be (as it seems), maybe  we can just re-use STM32F3 Kconfig options
...and maybe this whole file (if same exact same code is used)

**Review by erwango — CHANGES_REQUESTED** [review]
Some changes required to clock control and pinmux.

**Comment by galak** [comment]
I'm going to close #1103 since this takes it over.

File: boards/arm/nucleo_f030r8/pinmux.c (hunk 1 above)
Comment (superna9999) [reply boards/arm/nucleo_f030r8/pinmux.c]: Done, aligned Doc, Pinmux and DT for USART1

File: drivers/clock_control/stm32f0x_ll_clock.c (hunk 2 above)
Comment (superna9999) [reply drivers/clock_control/stm32f0x_ll_clock.c]: This is added in the stm32f091 support from @b0661, I will pick his change for that.

Comment (superna9999) [reply drivers/clock_control/stm32f0x_ll_clock.c]: Done, I merged it in the clock support patch

**Review by dbkinder — APPROVED** [review]

File: drivers/clock_control/stm32f0x_ll_clock.c (hunk 2 above)
Comment (erwango) [reply drivers/clock_control/stm32f0x_ll_clock.c]: Are there good reasons not to use stm32f3 file, as this is exact same code?
Do you foresee divergences in the future?

Comment (superna9999) [reply drivers/clock_control/stm32f0x_ll_clock.c]: I dont't have any opinion, should I update it ? the PR is ok to be merged, maybe it's something that can be cleaned up later on ?

Comment (erwango) [reply drivers/clock_control/stm32f0x_ll_clock.c]: ok, let's keep it that way for now. With time, we'll know if this is a good thing to do.

**Review by erwango — APPROVED** [review]

**Comment by galak** [comment]
Can you guys explain 'arch: arm: core: fix vector table relocate write to flash' to me a bit more.

**Comment by superna9999** [comment]
@galak On Cortex-M0, you cannot change the vector table pointer in the CPU registers, you need to have them at address 0 in physical memory, but on STM32 address 0 can be remapped on flash or whatever, so if we need to relocate the pointers we must be sure the DRAM is mapped here and reserve the memory in the binary ram sections

**Comment by galak** [comment]
If you can rebase and take of the conflict, I'll merge after that.

**Review by galak — APPROVED** [review]

**Comment by superna9999** [comment]
Ok I’ll do it ASAP

**Comment by galak** [comment]
Oh, forgot to mention that we need to look at tests/kernel/common/test, fails due to needing a few more bytes of SRAM.

**Pull request merged**